commons/src/main/java/org/infinispan/commons/marshall/MarshallUtil.java

```diff
@@ -41,4 +41,28 @@ public static void unmarshallMap(Map<Object, Object> map, ObjectInput in) throws
       int size = UnsignedNumeric.readUnsignedInt(in);
       for (int i = 0; i < size; i++) map.put(in.readObject(), in.readObject());
    }
+
+   public static <E extends Enum<E>> void marshallEnum(E e, ObjectOutput output, boolean checkNull) throws IOException {
+      if (checkNull && checkIfNull(e, output)) {
+         return;
+      }
+      output.writeByte(e.ordinal());
+   }
+
+   public static <E extends Enum<E>> E unmarshallEnum(ObjectInput input, E[] values, boolean checkNull) throws IOException {
+      if (checkNull && input.readBoolean()) {
+         return null;
+      }
+      try {
+         return values[input.readByte()];
+      } catch (ArrayIndexOutOfBoundsException e) {
+         throw new IOException("Unknown enum.", e);
+      }
+   }
+
+   private static boolean checkIfNull(Object obj, ObjectOutput output) throws IOException {
+      final boolean isNull = obj == null;
+      output.writeBoolean(isNull);
```

Comment:
I'm pretty sure JBoss Marshalling takes care of the `null` case by itself, because it couldn't pick the right externalizer for a `null` value, so we could just throw an exception here.